# Install the Watson IoT Platform Python SDK

In [1]:
pip install wiotp.sdk

     |████████████████████████████████| 96 kB 9.1 MB/s  eta 0:00:01
     |████████████████████████████████| 99 kB 8.7 MB/s  eta 0:00:01
     |████████████████████████████████| 54 kB 3.9 MB/s  eta 0:00:01
  Created wheel for wiotp.sdk: filename=wiotp_sdk-0.11.0-py3-none-any.whl size=97104 sha256=66f991be59fe4c5c51d14980ef8d6f8c3c99f3b8f233f5d7ee99fe6693b8ac2a
  Stored in directory: /tmp/wsuser/.cache/pip/wheels/56/f1/4b/2d7bc55ae8f9ef6e175c2c8be828fcd108b8c4b8c5ff613136
  Created wheel for paho-mqtt: filename=paho_mqtt-1.6.1-py3-none-any.whl size=62116 sha256=552f902820495896354180e5f871a4fe7a0a2a170dbf95582b8d33b14106de43
  Stored in directory: /tmp/wsuser/.cache/pip/wheels/d0/bf/ac/2b3f43f8c6fcd0f4ba5395397458c521eb0b52d33b574a5a40
Successfully built wiotp.sdk paho-mqtt
Note: you may need to restart the kernel to use updated packages.


# Import the (few) required packages

In [2]:
import wiotp.sdk.device
import pandas as pd
import time
import argparse
from datetime import datetime

# Capture your WIoTP Device Configuration

## OPTION A - For MONITOR SaaS

In [3]:
# OPTION A: For MONITOR SaaS

import wiotp.sdk.device

myConfig = { 
    "identity": {
        "orgId": "YOUR_ORGID",
        "typeId": "YOUR_Device",
        "deviceId": "YOUR_Device001"
    },
    "auth": {
        "token": "YOUR_AUTHENTICATION_TOKEN"
    }
}

## OPTION B - For MONITOR part of MAS (Maximo Application Suite)

In [3]:
# OPTION B: For MONITOR on MAS (Maximop Application Suite)

import wiotp.sdk.device

# Provide the Device's configuration
# Refer to for details on how to get all this info
myConfig = { 
    "identity": {
        "orgId": "YOUR_ORGID",
        "typeId": "YOUR_Device",
        "deviceId": "YOUR_Device001"
    },
    "auth": {
        "token": "YOUR_AUTHENTICATION_TOKEN"
    },
    "options": {
        "domain": "iot.maximoabc.yourcloud.info",
        "http":{
            "verify": "True"
        },
        "mqtt":{
            "port":443,
            "caFile": "/project_data/data_asset/iot-maximoabc-yourcloud-info-chain.pem"
        }
    }
}

# Connect your Device to WIoTP

In [4]:
deviceCli = wiotp.sdk.device.DeviceClient(config=myConfig)
print(myConfig)
deviceCli.connect()

2022-01-04 01:51:20,109   wiotp.sdk.device.client.DeviceClient  INFO    Connected successfully: d:YOUR_ORGID:YOUR_Device:YOUR_Device001


{'identity': {'orgId': 'YOUR_ORGID', 'typeId': 'YOUR_Device', 'deviceId': 'YOUR_Device001'}, 'auth': {'token': 'YOUR_AUTHENTICATION_TOKEN'}, 'options': {'domain': 'iot.maximoabc.yourcloud.info', 'http': {'verify': 'True'}, 'mqtt': {'port': 443, 'caFile': '/project_data/data_asset/iot-maximoabc-yourcloud-info-chain.pem', 'transport': 'tcp', 'cleanStart': False, 'sessionExpiry': 3600, 'keepAlive': 60}, 'logLevel': 20}}


# Read the '0_1080_All.csv' Pattern File

In [5]:
monitor_patterns_readings = pd.read_csv('/project_data/data_asset/0_1080_All.csv',
                 index_col=False)
monitor_patterns_readings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1080 entries, 0 to 1079
Data columns (total 7 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   time       1080 non-null   int64  
 1   reading_1  1080 non-null   float64
 2   reading_2  1080 non-null   float64
 3   reading_3  1080 non-null   float64
 4   reading_4  1080 non-null   float64
 5   reading_5  1080 non-null   float64
 6   reading_6  1080 non-null   float64
dtypes: float64(6), int64(1)
memory usage: 59.2 KB


# Send the 1080 rows of '0_1080_All.csv' data to your Device on WIoTP

In [6]:
for index, row in monitor_patterns_readings.iterrows():
    data = {'reading_1': row['reading_1'],
            'reading_2': row['reading_2'],
            'reading_3': row['reading_3'],
            'reading_4': row['reading_4'],
            'reading_5': row['reading_5'],
            'reading_6': row['reading_6']
            }
    print(data)

    parser = argparse.ArgumentParser()
    parser.add_argument("-E", "--event", required=False, default="event", help="type of event to send")
    parser.add_argument("-N", "--nummsgs", required=False, type=int, default=1, help="send this many messages before disconnecting")
    
    parser.add_argument("-D", "--delay", required=False, type=float, default=5, help="number of seconds between msgs")
    args, unknown = parser.parse_known_args()

    def myOnPublishCallback():
        print()

    success = deviceCli.publishEvent(args.event, "json", data, qos=0, onPublish=myOnPublishCallback)
    now = datetime.now()
    print("======SUCCESS======", now, "===Row:", index,"Success:", success)
    if not success:
        print("Not connected to WIoTP")

    time.sleep(args.delay)
    

deviceCli.disconnect()

{'reading_1': 10.85066167, 'reading_2': 10.01964812, 'reading_3': 20.15516953, 'reading_4': 20.87776326, 'reading_5': 10.04102436, 'reading_6': 12.94680491}
======SUCCESS====== 2022-01-04 01:52:23.098698 ===Row: 0 Success: True

{'reading_1': 10.59574274, 'reading_2': 10.80853745, 'reading_3': 20.87205928, 'reading_4': 20.10878476, 'reading_5': 10.17112379, 'reading_6': 10.32845991}
======SUCCESS====== 2022-01-04 01:52:28.122078 ===Row: 1 Success: True

{'reading_1': 10.07333239, 'reading_2': 10.48054819, 'reading_3': 20.02722103, 'reading_4': 20.46039852, 'reading_5': 10.35664339, 'reading_6': 13.93976121}
======SUCCESS====== 2022-01-04 01:52:33.149557 ===Row: 2 Success: True

{'reading_1': 10.90152952, 'reading_2': 10.21132138, 'reading_3': 20.57671781, 'reading_4': 20.21359487, 'reading_5': 10.95080441, 'reading_6': 10.69124238}
======SUCCESS====== 2022-01-04 01:52:38.176823 ===Row: 3 Success: True

{'reading_1': 10.54842036, 'reading_2': 10.4896714, 'reading_3': 20.52216801, 'readi

{'reading_1': 10.52234547, 'reading_2': 10.36484044, 'reading_3': 20.08028765, 'reading_4': 20.04705393, 'reading_5': 10.11678553, 'reading_6': 13.1292419}
======SUCCESS====== 2022-01-04 01:55:23.951829 ===Row: 36 Success: True

{'reading_1': 10.20136899, 'reading_2': 10.00304212, 'reading_3': 20.48567865, 'reading_4': 20.37314922, 'reading_5': 10.14345842, 'reading_6': 10.45554465}
======SUCCESS====== 2022-01-04 01:55:28.971654 ===Row: 37 Success: True

{'reading_1': 10.80886107, 'reading_2': 10.74164976, 'reading_3': 20.25778963, 'reading_4': 20.24706856, 'reading_5': 10.511980300000001, 'reading_6': 13.12175555}
======SUCCESS====== 2022-01-04 01:55:33.998015 ===Row: 38 Success: True

{'reading_1': 10.76131408, 'reading_2': 10.02212994, 'reading_3': 20.06920683, 'reading_4': 20.58498817, 'reading_5': 10.98114737, 'reading_6': 10.19189523}
======SUCCESS====== 2022-01-04 01:55:39.024570 ===Row: 39 Success: True

{'reading_1': 10.59287702, 'reading_2': 10.70983011, 'reading_3': 20.76562

{'reading_1': 10.96233912, 'reading_2': 11.168484900000001, 'reading_3': 23.88836232, 'reading_4': 23.96212364, 'reading_5': 10.57013926, 'reading_6': 12.61549734}
======SUCCESS====== 2022-01-04 01:58:24.797104 ===Row: 72 Success: True

{'reading_1': 10.40360585, 'reading_2': 11.11193032, 'reading_3': 23.17512971, 'reading_4': 23.99251421, 'reading_5': 10.86178183, 'reading_6': 11.66367074}
======SUCCESS====== 2022-01-04 01:58:29.823957 ===Row: 73 Success: True

{'reading_1': 10.14727465, 'reading_2': 10.854824699999998, 'reading_3': 23.5403244, 'reading_4': 23.12849283, 'reading_5': 10.49158549, 'reading_6': 13.36904021}
======SUCCESS====== 2022-01-04 01:58:34.842659 ===Row: 74 Success: True

{'reading_1': 10.77896673, 'reading_2': 11.57355516, 'reading_3': 23.33930739, 'reading_4': 23.368385, 'reading_5': 10.27672537, 'reading_6': 11.58706602}
======SUCCESS====== 2022-01-04 01:58:39.869125 ===Row: 75 Success: True

{'reading_1': 10.007681800000002, 'reading_2': 11.58053225, 'reading_

{'reading_1': 10.23419331, 'reading_2': 11.83284581, 'reading_3': 25.06270413, 'reading_4': 25.04186712, 'reading_5': 20.1414503, 'reading_6': 12.62073144}
======SUCCESS====== 2022-01-04 02:01:25.673588 ===Row: 108
 Success: True
{'reading_1': 10.33164213, 'reading_2': 11.42243306, 'reading_3': 25.52773385, 'reading_4': 25.48888595, 'reading_5': 20.92472351, 'reading_6': 11.10433185}
======SUCCESS====== 2022-01-04 02:01:30.699375 ===Row: 109 Success: True

{'reading_1': 10.58394862, 'reading_2': 11.51120221, 'reading_3': 25.69741381, 'reading_4': 25.67096553, 'reading_5': 20.188368899999997, 'reading_6': 12.71003691}
======SUCCESS====== 2022-01-04 02:01:35.726010 ===Row: 110 Success: True

{'reading_1': 10.80605891, 'reading_2': 11.80461587, 'reading_3': 25.82966748, 'reading_4': 25.40522407, 'reading_5': 20.58918726, 'reading_6': 11.51992034}
======SUCCESS====== 2022-01-04 02:01:40.751664 ===Row: 111 Success: True

{'reading_1': 10.90678698, 'reading_2': 11.45640713, 'reading_3': 25.3

{'reading_1': 10.91532854, 'reading_2': 12.98391139, 'reading_3': 25.36291096, 'reading_4': 25.70346445, 'reading_5': 20.3743336, 'reading_6': 12.23779786}
======SUCCESS====== 2022-01-04 02:04:26.548839 ===Row: 144 Success: True

{'reading_1': 10.37076753, 'reading_2': 12.14934757, 'reading_3': 25.94981437, 'reading_4': 25.62703876, 'reading_5': 20.49636707, 'reading_6': 11.51899891}
======SUCCESS====== 2022-01-04 02:04:31.575473 ===Row: 145 Success: True

{'reading_1': 10.71503565, 'reading_2': 12.60543561, 'reading_3': 25.178634, 'reading_4': 25.9558296, 'reading_5': 20.96593918, 'reading_6': 12.60597666}
======SUCCESS====== 2022-01-04 02:04:36.589647 ===Row: 146 
Success: True
{'reading_1': 10.71374828, 'reading_2': 12.62617027, 'reading_3': 25.95287545, 'reading_4': 25.79256005, 'reading_5': 20.20300859, 'reading_6': 11.12760022}
======SUCCESS====== 2022-01-04 02:04:41.615753 ===Row: 147 Success: True

{'reading_1': 10.80528358, 'reading_2': 12.80008492, 'reading_3': 25.91586625, '

{'reading_1': 10.85694017, 'reading_2': 14.87132399, 'reading_3': 21.73415858, 'reading_4': 21.30646471, 'reading_5': 20.20756138, 'reading_6': 12.67045496}
======SUCCESS====== 2022-01-04 02:07:27.375173 ===Row: 180 Success: True

{'reading_1': 10.92514598, 'reading_2': 14.333857, 'reading_3': 21.38409247, 'reading_4': 21.81826229, 'reading_5': 20.25385302, 'reading_6': 11.49172676}
======SUCCESS====== 2022-01-04 02:07:32.402351 ===Row: 181 Success: True

{'reading_1': 10.63502773, 'reading_2': 14.66684405, 'reading_3': 21.18747723, 'reading_4': 21.60526039, 'reading_5': 20.73159933, 'reading_6': 12.37009576}
======SUCCESS====== 2022-01-04 02:07:37.428996 ===Row: 182 Success: True

{'reading_1': 10.5352476, 'reading_2': 14.11907827, 'reading_3': 21.78851162, 'reading_4': 21.82301466, 'reading_5': 20.12944099, 'reading_6': 11.04358492}
======SUCCESS====== 2022-01-04 02:07:42.455111 ===Row: 183 Success: True

{'reading_1': 10.36256993, 'reading_2': 14.88503997, 'reading_3': 21.45196173, 

{'reading_1': 7.401371266, 'reading_2': 15.32517114, 'reading_3': 20.3974791, 'reading_4': 20.82070684, 'reading_5': 10.63336273, 'reading_6': 12.135682300000001}
======SUCCESS====== 2022-01-04 02:10:28.305002 ===Row: 216 Success: True

{'reading_1': 7.651083726, 'reading_2': 15.12307744, 'reading_3': 20.84276976, 'reading_4': 20.97935917, 'reading_5': 10.47363489, 'reading_6': 11.89854359}

======SUCCESS====== 2022-01-04 02:10:33.331312 ===Row: 217 Success: True
{'reading_1': 7.280353186, 'reading_2': 15.15021175, 'reading_3': 20.84440377, 'reading_4': 20.42177844, 'reading_5': 10.00637747, 'reading_6': 12.46086839}
======SUCCESS====== 2022-01-04 02:10:38.356382 ===Row: 218 Success: True

{'reading_1': 7.38915317, 'reading_2': 15.76423262, 'reading_3': 20.757704999999998, 'reading_4': 20.36313935, 'reading_5': 10.06309551, 'reading_6': 11.56087467}
======SUCCESS====== 2022-01-04 02:10:43.380721 ===Row: 219 Success: True

{'reading_1': 7.524389948, 'reading_2': 15.19420025, 'reading_3'

{'reading_1': 10.5244547, 'reading_2': 15.22249161, 'reading_3': 21.73197131, 'reading_4': 21.47004183, 'reading_5': 10.80145449, 'reading_6': 12.49488694}
======SUCCESS====== 2022-01-04 02:13:29.153350 ===Row: 252 Success: True

{'reading_1': 10.40560505, 'reading_2': 15.31564034, 'reading_3': 21.31428611, 'reading_4': 21.68334952, 'reading_5': 10.01668926, 'reading_6': 11.84584705}
======SUCCESS====== 2022-01-04 02:13:34.178414 ===Row: 253 Success: True

{'reading_1': 10.64480896, 'reading_2': 15.89912631, 'reading_3': 21.61321436, 'reading_4': 21.48767994, 'reading_5': 10.32910906, 'reading_6': 12.24128304}
======SUCCESS====== 2022-01-04 02:13:39.204517 ===Row: 254 Success: True

{'reading_1': 10.67624153, 'reading_2': 15.83328714, 'reading_3': 21.18251557, 'reading_4': 21.81022876, 'reading_5': 10.67962314, 'reading_6': 11.51621272}
======SUCCESS====== 2022-01-04 02:13:44.222557 ===Row: 255 Success: True

{'reading_1': 10.22828855, 'reading_2': 15.49994898, 'reading_3': 21.63276555

{'reading_1': 10.86818426, 'reading_2': 15.21394165, 'reading_3': 24.33647792, 'reading_4': 24.40538695, 'reading_5': 10.46084667, 'reading_6': 12.35020941}
======SUCCESS====== 2022-01-04 02:16:29.982679 ===Row: 288 Success: True

{'reading_1': 10.68400355, 'reading_2': 15.45599458, 'reading_3': 24.35050628, 'reading_4': 24.94614669, 'reading_5': 10.09511802, 'reading_6': 11.96337173}

======SUCCESS====== 2022-01-04 02:16:35.005648 ===Row: 289 Success: True
{'reading_1': 10.92486061, 'reading_2': 15.89592353, 'reading_3': 25.25361211, 'reading_4': 24.54418964, 'reading_5': 10.10673801, 'reading_6': 12.33432622}
======SUCCESS====== 2022-01-04 02:16:40.019730 ===Row: 290 Success: True

{'reading_1': 10.02369344, 'reading_2': 15.71026366, 'reading_3': 24.88085806, 'reading_4': 24.56228338, 'reading_5': 10.50860621, 'reading_6': 11.88326933}
======SUCCESS====== 2022-01-04 02:16:45.046094 ===Row: 291 Success: True

{'reading_1': 10.6318314, 'reading_2': 15.76118698, 'reading_3': 24.51199292

{'reading_1': 10.94827671, 'reading_2': 14.33514711, 'reading_3': 25.8574091, 'reading_4': 25.71486822, 'reading_5': 30.93013353, 'reading_6': 12.0}
======SUCCESS====== 2022-01-04 02:19:30.875068 ===Row: 324 Success: True

{'reading_1': 10.00308158, 'reading_2': 13.62080147, 'reading_3': 25.42596467, 'reading_4': 25.11691986, 'reading_5': 30.75412241, 'reading_6': 12.0}
======SUCCESS====== 2022-01-04 02:19:35.901811 ===Row: 325 Success: True

{'reading_1': 10.73310856, 'reading_2': 13.85333521, 'reading_3': 25.83864438, 'reading_4': 25.52492538, 'reading_5': 30.10492633, 'reading_6': 12.0}
======SUCCESS====== 2022-01-04 02:19:40.928034 ===Row: 326 Success: True

{'reading_1': 10.58571155, 'reading_2': 13.71808767, 'reading_3': 25.54529787, 'reading_4': 25.44750053, 'reading_5': 30.39468946, 'reading_6': 12.0}
======SUCCESS====== 2022-01-04 02:19:45.954055 ===Row: 327 Success: True

{'reading_1': 10.70925014, 'reading_2': 13.42214636, 'reading_3': 25.39900801, 'reading_4': 25.41192199, 

{'reading_1': 12.97750733, 'reading_2': 12.60219698, 'reading_3': 23.19996237, 'reading_4': 23.49485927, 'reading_5': 30.73551856, 'reading_6': 10.21108733}
======SUCCESS====== 2022-01-04 02:22:36.753685 ===Row: 361 Success: True

{'reading_1': 12.86226959, 'reading_2': 12.11796224, 'reading_3': 23.63333462, 'reading_4': 23.81887377, 'reading_5': 30.98452966, 'reading_6': 12.06755244}
======SUCCESS====== 2022-01-04 02:22:41.776575 ===Row: 362 Success: True

{'reading_1': 12.42293716, 'reading_2': 12.91835883, 'reading_3': 23.88341176, 'reading_4': 23.7892218, 'reading_5': 30.11999891, 'reading_6': 11.5235257}
======SUCCESS====== 2022-01-04 02:22:46.804994 ===Row: 363 Success: True

{'reading_1': 12.34311962, 'reading_2': 12.39715043, 'reading_3': 23.22752752, 'reading_4': 23.63514282, 'reading_5': 30.40153549, 'reading_6': 13.88069971}
======SUCCESS====== 2022-01-04 02:22:51.832062 ===Row: 364 Success: True

{'reading_1': 12.49985697, 'reading_2': 12.45143994, 'reading_3': 23.50180282,

{'reading_1': 10.61241817, 'reading_2': 11.07343933, 'reading_3': 28.55944657, 'reading_4': 23.0, 'reading_5': 30.3705218, 'reading_6': 13.05686732}
======SUCCESS====== 2022-01-04 02:25:42.655466 ===Row: 398 Success: True

{'reading_1': 10.89193215, 'reading_2': 10.23870938, 'reading_3': 28.68702567, 'reading_4': 23.0, 'reading_5': 10.14744727, 'reading_6': 10.42303537}
======SUCCESS====== 2022-01-04 02:25:47.675792 ===Row: 399 Success: True

{'reading_1': 10.02149745, 'reading_2': 10.21442713, 'reading_3': 25.62719923, 'reading_4': 23.0, 'reading_5': 10.99929167, 'reading_6': 13.89594403}
======SUCCESS====== 2022-01-04 02:25:52.698423 ===Row: 400 Success: True

{'reading_1': 10.07407046, 'reading_2': 10.62244849, 'reading_3': 25.82300811, 'reading_4': 23.0, 'reading_5': 10.06284743, 'reading_6': 10.79465827}
======SUCCESS====== 2022-01-04 02:25:57.725782 ===Row: 401 Success: True

{'reading_1': 10.79836374, 'reading_2': 10.67739866, 'reading_3': 25.64092625, 'reading_4': 23.0, 'readin

{'reading_1': 10.92779132, 'reading_2': 10.42807439, 'reading_3': 16.84926402, 'reading_4': 23.0, 'reading_5': 10.16037031, 'reading_6': 11.36279803}
======SUCCESS====== 2022-01-04 02:28:48.569056 ===Row: 435 Success: True

{'reading_1': 10.35649213, 'reading_2': 10.70180074, 'reading_3': 16.16778479, 'reading_4': 23.0, 'reading_5': 10.43563572, 'reading_6': 12.41350915}
======SUCCESS====== 2022-01-04 02:28:53.591474 ===Row: 436 Success: True

{'reading_1': 10.538089300000001, 'reading_2': 10.59290485, 'reading_3': 16.91775909, 'reading_4': 23.0, 'reading_5': 10.38578272, 'reading_6': 10.1012107}
======SUCCESS====== 2022-01-04 02:28:58.617209 ===Row: 437 Success: True

{'reading_1': 10.13215181, 'reading_2': 10.71379935, 'reading_3': 16.80709313, 'reading_4': 23.0, 'reading_5': 10.1458028, 'reading_6': 12.96143774}
======SUCCESS====== 2022-01-04 02:29:03.643999 ===Row: 438 Success: True

{'reading_1': 10.452808500000002, 'reading_2': 10.49881854, 'reading_3': 16.43151884, 'reading_4': 

{'reading_1': 10.15944504, 'reading_2': 10.04257531, 'reading_3': 32.25158293, 'reading_4': 23.16712289, 'reading_5': 10.21430168, 'reading_6': 12.48381553}
======SUCCESS====== 2022-01-04 02:31:54.452714 ===Row: 472 Success: True

{'reading_1': 10.98961008, 'reading_2': 10.01350779, 'reading_3': 32.57004557, 'reading_4': 23.75299237, 'reading_5': 10.81591545, 'reading_6': 11.77732274}
======SUCCESS====== 2022-01-04 02:31:59.477834 ===Row: 473 Success:
 True
{'reading_1': 10.88175455, 'reading_2': 10.89037667, 'reading_3': 32.50050425, 'reading_4': 23.44923864, 'reading_5': 10.42286219, 'reading_6': 12.19069721}
======SUCCESS====== 2022-01-04 02:32:04.503078 ===Row: 474 Success: True

{'reading_1': 10.964666000000001, 'reading_2': 10.4800319, 'reading_3': 32.16341022, 'reading_4': 23.53655186, 'reading_5': 10.95014707, 'reading_6': 11.69150292}
======SUCCESS====== 2022-01-04 02:32:09.529718 ===Row: 475 Success: True

{'reading_1': 10.54025528, 'reading_2': 10.70502031, 'reading_3': 32.2

{'reading_1': 11.56496292, 'reading_2': 10.668100099999998, 'reading_3': 22.80082973, 'reading_4': 25.65673557, 'reading_5': 40.38182826, 'reading_6': 12.44403471}
======SUCCESS====== 2022-01-04 02:34:55.306093 ===Row: 508 Success: True

{'reading_1': 11.50992399, 'reading_2': 11.03397193, 'reading_3': 22.72581893, 'reading_4': 25.4316326, 'reading_5': 40.05051999, 'reading_6': 11.57471961}
======SUCCESS====== 2022-01-04 02:35:00.320485 ===Row: 509 Success: True

{'reading_1': 10.95607089, 'reading_2': 10.47825176, 'reading_3': 20.52175811, 'reading_4': 25.26416929, 'reading_5': 40.3043019, 'reading_6': 12.41329436}
======SUCCESS====== 2022-01-04 02:35:05.347127 ===Row: 510 Success: True

{'reading_1': 11.4745588, 'reading_2': 11.28275628, 'reading_3': 20.5730708, 'reading_4': 25.03269225, 'reading_5': 40.696408, 'reading_6': 11.68044924}
======SUCCESS====== 2022-01-04 02:35:10.369119 ===Row: 511 Success: True

{'reading_1': 11.66752294, 'reading_2': 11.11589706, 'reading_3': 19.210915

{'reading_1': 10.9737744, 'reading_2': 11.17412235, 'reading_3': 24.36354149, 'reading_4': 25.31571886, 'reading_5': 40.12926606, 'reading_6': 12.02375554}
======SUCCESS====== 2022-01-04 02:37:56.178115 ===Row: 544 Success: True

{'reading_1': 10.80996481, 'reading_2': 11.657072099999999, 'reading_3': 24.58621663, 'reading_4': 25.283559699999998, 'reading_5': 40.71780775, 'reading_6': 11.74171145}
======SUCCESS====== 2022-01-04 02:38:01.205223 ===Row: 545 Success: True

{'reading_1': 10.18629963, 'reading_2': 11.88035118, 'reading_3': 24.91008544, 'reading_4': 25.61775763, 'reading_5': 40.27155357, 'reading_6': 12.11961679}
======SUCCESS====== 2022-01-04 02:38:06.232272 ===Row: 546 Success: True

{'reading_1': 10.57026067, 'reading_2': 11.64782613, 'reading_3': 24.68233661, 'reading_4': 25.48733751, 'reading_5': 40.51115755, 'reading_6': 11.8371517}
======SUCCESS====== 2022-01-04 02:38:11.254746 ===Row: 547 Success: True

{'reading_1': 10.85547909, 'reading_2': 11.34942493, 'reading_3'

{'reading_1': 10.53978048, 'reading_2': 12.60564908, 'reading_3': 21.11440331, 'reading_4': 21.58579643, 'reading_5': 40.36847442, 'reading_6': 11.0}
======SUCCESS====== 2022-01-04 02:41:02.047125 ===Row: 581 Success: True

{'reading_1': 10.02645831, 'reading_2': 12.91455641, 'reading_3': 21.12446124, 'reading_4': 21.06180532, 'reading_5': 40.50870245, 'reading_6': 11.0}
======SUCCESS====== 2022-01-04 02:41:07.071037 ===Row: 582 Success: True

{'reading_1': 10.77965356, 'reading_2': 12.57746011, 'reading_3': 21.27206542, 'reading_4': 21.54017794, 'reading_5': 40.65129263, 'reading_6': 11.0}
======SUCCESS====== 2022-01-04 02:41:12.097443 ===Row: 583 Success: True

{'reading_1': 10.29449336, 'reading_2': 12.59546304, 'reading_3': 21.41925422, 'reading_4': 21.80386587, 'reading_5': 40.97530203, 'reading_6': 11.0}
======SUCCESS====== 2022-01-04 02:41:17.124650 ===Row: 584 Success: True

{'reading_1': 10.16430444, 'reading_2': 12.12208717, 'reading_3': 21.268606600000002, 'reading_4': 21.90

{'reading_1': 10.0895153, 'reading_2': 13.51137062, 'reading_3': 20.76641363, 'reading_4': 20.16900612, 'reading_5': 10.89969919, 'reading_6': 11.26575544}
======SUCCESS====== 2022-01-04 02:44:02.982869 ===Row: 617 Success: True

{'reading_1': 10.68440722, 'reading_2': 13.04319174, 'reading_3': 20.7048178, 'reading_4': 20.97810628, 'reading_5': 10.72422072, 'reading_6': 10.68527532}
======SUCCESS====== 2022-01-04 02:44:08.009111 ===Row: 618 Success: True

{'reading_1': 10.69090103, 'reading_2': 13.58037532, 'reading_3': 20.83785139, 'reading_4': 20.17377689, 'reading_5': 10.04376695, 'reading_6': 11.29922604}
======SUCCESS====== 2022-01-04 02:44:13.034768 ===Row: 619 Success: True

{'reading_1': 10.36379657, 'reading_2': 13.59588449, 'reading_3': 20.34954274, 'reading_4': 20.4627867, 'reading_5': 10.57899779, 'reading_6': 10.78748076}
======SUCCESS====== 2022-01-04 02:44:18.059457 ===Row: 620 Success: True

{'reading_1': 10.14077238, 'reading_2': 13.63334742, 'reading_3': 20.92752745, 

{'reading_1': 10.19413444, 'reading_2': 15.853555100000001, 'reading_3': 21.41040192, 'reading_4': 21.47994383, 'reading_5': 10.88722904, 'reading_6': 11.11585257}
======SUCCESS====== 2022-01-04 02:47:03.805408 ===Row: 653 Success: True

{'reading_1': 9.979107159, 'reading_2': 15.32418971, 'reading_3': 21.85746845, 'reading_4': 21.28114062, 'reading_5': 10.15069546, 'reading_6': 10.55265757}
======SUCCESS====== 2022-01-04 02:47:08.831734 ===Row: 654 Success: True

{'reading_1': 9.530652407, 'reading_2': 15.54455211, 'reading_3': 21.71346365, 'reading_4': 21.53630076, 'reading_5': 10.26814182, 'reading_6': 11.22187341}
======SUCCESS====== 2022-01-04 02:47:13.858508 ===Row: 655 Success: True

{'reading_1': 10.00068538, 'reading_2': 15.07761378, 'reading_3': 21.03678804, 'reading_4': 21.62028645, 'reading_5': 10.67736551, 'reading_6': 10.57757814}
======SUCCESS====== 2022-01-04 02:47:18.885407 ===Row: 656 Success: True

{'reading_1': 10.23666864, 'reading_2': 14.96302319, 'reading_3': 21.

{'reading_1': 10.85524898, 'reading_2': 15.9996226, 'reading_3': 24.35137429, 'reading_4': 24.75953152, 'reading_5': 10.5621235, 'reading_6': 11.40006597}
======SUCCESS====== 2022-01-04 02:50:04.703003 ===Row: 689 Success: True

{'reading_1': 10.94616531, 'reading_2': 15.0002196, 'reading_3': 25.31838212, 'reading_4': 25.056811899999996, 'reading_5': 10.73387171, 'reading_6': 10.86581899}
======SUCCESS====== 2022-01-04 02:50:09.725330 ===Row: 690 Success: True

{'reading_1': 10.78691209, 'reading_2': 15.15640455, 'reading_3': 24.7356163, 'reading_4': 24.77212571, 'reading_5': 10.78279821, 'reading_6': 11.47918586}
======SUCCESS====== 2022-01-04 02:50:14.752563 ===Row: 691 Success: True

{'reading_1': 10.22264079, 'reading_2': 15.78369021, 'reading_3': 25.10780618, 'reading_4': 25.01525786, 'reading_5': 10.10251093, 'reading_6': 10.80597971}
======SUCCESS====== 2022-01-04 02:50:19.777927 ===Row: 692 Success: True

{'reading_1': 10.50090445, 'reading_2': 15.85299432, 'reading_3': 24.7372

{'reading_1': 10.43192926, 'reading_2': 15.31203806, 'reading_3': 25.47636371, 'reading_4': 25.82832406, 'reading_5': 50.74115002, 'reading_6': 10.55928909}
======SUCCESS====== 2022-01-04 02:53:05.592195 ===Row: 725 Success: True

{'reading_1': 10.43436064, 'reading_2': 15.15780988, 'reading_3': 25.55531111, 'reading_4': 25.99891773, 'reading_5': 50.70621976, 'reading_6': 11.85886352}
======SUCCESS====== 2022-01-04 02:53:10.618704 ===Row: 726 Success: True

{'reading_1': 10.71918722, 'reading_2': 15.32872931, 'reading_3': 25.77155764, 'reading_4': 25.94874221, 'reading_5': 50.81491841, 'reading_6': 10.34250365}
======SUCCESS====== 2022-01-04 02:53:15.645052 ===Row: 727 Success: True

{'reading_1': 10.59772859, 'reading_2': 15.74572823, 'reading_3': 25.16421264, 'reading_4': 25.99218485, 'reading_5': 50.76860754, 'reading_6': 11.35804835}
======SUCCESS====== 2022-01-04 02:53:20.671632 ===Row: 728 Success: True

{'reading_1': 10.11649046, 'reading_2': 15.07630953, 'reading_3': 25.0688332

{'reading_1': 10.794267999999999, 'reading_2': 14.85763795, 'reading_3': 23.81974186, 'reading_4': 23.93421809, 'reading_5': 50.24300544, 'reading_6': 10.30561753}
======SUCCESS====== 2022-01-04 02:56:06.485083 ===Row: 761 Success: True

{'reading_1': 10.56394266, 'reading_2': 14.64519447, 'reading_3': 23.74134024, 'reading_4': 23.94112012, 'reading_5': 50.18295043, 'reading_6': 11.94210786}
======SUCCESS====== 2022-01-04 02:56:11.508662 ===Row: 762 Success: True

{'reading_1': 10.10789607, 'reading_2': 14.88083755, 'reading_3': 23.75641569, 'reading_4': 23.06865218, 'reading_5': 50.07112928, 'reading_6': 10.85420282}
======SUCCESS====== 2022-01-04 02:56:16.526675 ===Row: 763 Success: True

{'reading_1': 10.64575039, 'reading_2': 14.72701005, 'reading_3': 23.97275661, 'reading_4': 23.24301277, 'reading_5': 50.35464233, 'reading_6': 11.52084062}
======SUCCESS====== 2022-01-04 02:56:21.553118 ===Row: 764 Success: True

{'reading_1': 10.11622647, 'reading_2': 14.83847503, 'reading_3': 23.

{'reading_1': 10.66092177, 'reading_2': 13.14309462, 'reading_3': 20.92828974, 'reading_4': 21.14445944, 'reading_5': 50.82815515, 'reading_6': 10.3856001}
======SUCCESS====== 2022-01-04 02:59:07.312779 ===Row: 797 Success: True

{'reading_1': 10.80167226, 'reading_2': 12.72500625, 'reading_3': 20.7547612, 'reading_4': 21.18217068, 'reading_5': 50.12254464, 'reading_6': 11.68224828}
======SUCCESS====== 2022-01-04 02:59:12.341606 ===Row: 798 Success: True

{'reading_1': 10.73430632, 'reading_2': 13.37399394, 'reading_3': 20.87072125, 'reading_4': 20.33848812, 'reading_5': 40.44141379, 'reading_6': 10.77457498}
======SUCCESS====== 2022-01-04 02:59:17.351529 ===Row: 799 Success: True

{'reading_1': 10.76136711, 'reading_2': 12.32729548, 'reading_3': 20.00542295, 'reading_4': 20.16675192, 'reading_5': 40.83916637, 'reading_6': 11.80746478}
======SUCCESS====== 2022-01-04 02:59:22.377924 ===Row: 800 Success: True

{'reading_1': 10.74996917, 'reading_2': 12.94797298, 'reading_3': 20.6742538, 

{'reading_1': 10.72717816, 'reading_2': 11.57934431, 'reading_3': 20.24128017, 'reading_4': 20.14011265, 'reading_5': 40.09702855, 'reading_6': 10.35490534}
======SUCCESS====== 2022-01-04 03:02:08.183010 ===Row: 833 Success: True

{'reading_1': 10.2105867, 'reading_2': 11.40273393, 'reading_3': 20.20746994, 'reading_4': 20.54440725, 'reading_5': 40.44609299, 'reading_6': 11.10953761}
======SUCCESS====== 2022-01-04 03:02:13.207439 ===Row: 834 Success: True

{'reading_1': 10.23165568, 'reading_2': 11.06628079, 'reading_3': 20.31489137, 'reading_4': 20.98618123, 'reading_5': 40.73830183, 'reading_6': 9.773088946}
======SUCCESS====== 2022-01-04 03:02:18.234108 ===Row: 835 Success: True

{'reading_1': 10.222838099999999, 'reading_2': 11.65883103, 'reading_3': 20.20379816, 'reading_4': 20.09170851, 'reading_5': 40.9576195, 'reading_6': 12.34921448}
======SUCCESS====== 2022-01-04 03:02:23.261845 ===Row: 836 Success: True

{'reading_1': 10.70496875, 'reading_2': 14.41002501, 'reading_3': 20.92

{'reading_1': 10.94206254, 'reading_2': 10.79670605, 'reading_3': 22.47808742, 'reading_4': 22.56550414, 'reading_5': 40.15353551, 'reading_6': 9.469210917}
======SUCCESS====== 2022-01-04 03:05:09.076934 ===Row: 869 Success: True

{'reading_1': 10.51245807, 'reading_2': 10.39182816, 'reading_3': 23.33269226, 'reading_4': 23.27552414, 'reading_5': 40.49892144, 'reading_6': 11.81568179}

======SUCCESS====== 2022-01-04 03:05:14.104296 ===Row: 870 Success: True
{'reading_1': 10.23637364, 'reading_2': 10.66198145, 'reading_3': 23.25224702, 'reading_4': 23.45013204, 'reading_5': 40.85053017, 'reading_6': 10.94131916}
======SUCCESS====== 2022-01-04 03:05:19.130565 ===Row: 871 Success: True

{'reading_1': 10.78742917, 'reading_2': 10.40719529, 'reading_3': 23.05414824, 'reading_4': 23.59970878, 'reading_5': 40.7155135, 'reading_6': 11.67351021}
======SUCCESS====== 2022-01-04 03:05:24.139268 ===Row: 872 Success: True

{'reading_1': 10.55576148, 'reading_2': 10.94264772, 'reading_3': 23.11582664

{'reading_1': 10.90727679, 'reading_2': 10.10488004, 'reading_3': 25.43950189, 'reading_4': 25.66934216, 'reading_5': 30.01240076, 'reading_6': 11.71189708}
======SUCCESS====== 2022-01-04 03:08:09.919808 ===Row: 905 Success: True

{'reading_1': 10.72323772, 'reading_2': 10.31335551, 'reading_3': 25.339649100000003, 'reading_4': 25.35499983, 'reading_5': 30.04240424, 'reading_6': 12.21006112}
======SUCCESS====== 2022-01-04 03:08:14.945757 ===Row: 906 Success: True

{'reading_1': 10.73143802, 'reading_2': 10.8507626, 'reading_3': 25.20053568, 'reading_4': 25.2049586, 'reading_5': 30.20957683, 'reading_6': 11.85535959}
======SUCCESS====== 2022-01-04 03:08:19.948239 ===Row: 907 Success: True

{'reading_1': 10.31794321, 'reading_2': 10.23829821, 'reading_3': 25.22767098, 'reading_4': 25.46735944, 'reading_5': 30.37886384, 'reading_6': 12.11344774}
======SUCCESS====== 2022-01-04 03:08:24.971933 ===Row: 908 Success: True

{'reading_1': 10.68366286, 'reading_2': 10.66216863, 'reading_3': 25.41

{'reading_1': 10.54654148, 'reading_2': 10.60748047, 'reading_3': 25.34228713, 'reading_4': 25.63074918, 'reading_5': 30.85596362, 'reading_6': 11.88161331}
======SUCCESS====== 2022-01-04 03:11:10.750128 ===Row: 941 Success: True

{'reading_1': 10.14692796, 'reading_2': 10.34183612, 'reading_3': 25.15975432, 'reading_4': 25.37574162, 'reading_5': 30.62738267, 'reading_6': 12.44105733}
======SUCCESS====== 2022-01-04 03:11:15.777040 ===Row: 942 Success: True

{'reading_1': 10.82606382, 'reading_2': 10.51566749, 'reading_3': 25.06667212, 'reading_4': 25.25339095, 'reading_5': 30.7652516, 'reading_6': 11.66272026}
======SUCCESS====== 2022-01-04 03:11:20.802354 ===Row: 943 Success: True

{'reading_1': 10.64427996, 'reading_2': 10.446795, 'reading_3': 25.33857697, 'reading_4': 25.7403992, 'reading_5': 30.85077565, 'reading_6': 12.2396369}
======SUCCESS====== 2022-01-04 03:11:25.804316 ===Row: 944 Success: True

{'reading_1': 10.97726768, 'reading_2': 10.39637622, 'reading_3': 25.16471814, 'r

{'reading_1': 10.53309896, 'reading_2': 11.49108254, 'reading_3': 22.91861155, 'reading_4': 22.93982242, 'reading_5': 30.71337257, 'reading_6': 11.79059931}
======SUCCESS====== 2022-01-04 03:14:11.627827 ===Row: 977 Success: True

{'reading_1': 10.71248276, 'reading_2': 11.10577165, 'reading_3': 22.92000967, 'reading_4': 22.06431885, 'reading_5': 30.613278700000002, 'reading_6': 12.42541408}
======SUCCESS====== 2022-01-04 03:14:16.652111 ===Row: 978 Success: True

{'reading_1': 10.98196687, 'reading_2': 11.2023066, 'reading_3': 22.58886969, 'reading_4': 22.94508504, 'reading_5': 30.95302761, 'reading_6': 11.93025057}
======SUCCESS====== 2022-01-04 03:14:21.678584 ===Row: 979 Success: True

{'reading_1': 10.36888101, 'reading_2': 11.55435575, 'reading_3': 21.45356413, 'reading_4': 21.58893113, 'reading_5': 30.22639891, 'reading_6': 12.25640536}
======SUCCESS====== 2022-01-04 03:14:26.702161 ===Row: 980 Success: True

{'reading_1': 10.02176362, 'reading_2': 11.21472199, 'reading_3': 21.8

{'reading_1': 10.75069814, 'reading_2': 11.33528846, 'reading_3': 20.6285625, 'reading_4': 20.80478133, 'reading_5': 20.29866568, 'reading_6': 11.82586626}
======SUCCESS====== 2022-01-04 03:17:12.462686 ===Row: 1013 Success: True

{'reading_1': 10.81541729, 'reading_2': 11.83971116, 'reading_3': 20.95172463, 'reading_4': 20.00970745, 'reading_5': 20.57408218, 'reading_6': 12.29213774}

======SUCCESS====== 2022-01-04 03:17:17.489024 ===Row: 1014 Success: True
{'reading_1': 10.64325734, 'reading_2': 11.94034253, 'reading_3': 20.05399943, 'reading_4': 20.53600225, 'reading_5': 20.4000397, 'reading_6': 11.64632362}
======SUCCESS====== 2022-01-04 03:17:22.515491 ===Row: 1015 Success: True

{'reading_1': 10.26179831, 'reading_2': 11.89439302, 'reading_3': 20.79625505, 'reading_4': 20.63385495, 'reading_5': 20.20707105, 'reading_6': 12.23672471}
======SUCCESS====== 2022-01-04 03:17:27.542349 ===Row: 1016 Success: True

{'reading_1': 10.03429322, 'reading_2': 11.95877729, 'reading_3': 20.46502

{'reading_1': 10.18377834, 'reading_2': 13.04307804, 'reading_3': 20.74487692, 'reading_4': 20.78267975, 'reading_5': 20.42105831, 'reading_6': 11.7238748}
======SUCCESS====== 2022-01-04 03:20:13.320682 ===Row: 1049 Success: True

{'reading_1': 10.96816014, 'reading_2': 12.99154459, 'reading_3': 21.62745565, 'reading_4': 21.29397096, 'reading_5': 20.64185293, 'reading_6': 12.25200172}
======SUCCESS====== 2022-01-04 03:20:18.347041 ===Row: 1050 Success: True

{'reading_1': 10.09171376, 'reading_2': 13.05924698, 'reading_3': 21.11299262, 'reading_4': 21.74954799, 'reading_5': 20.46292602, 'reading_6': 11.61437012}
======SUCCESS====== 2022-01-04 03:20:23.373637 ===Row: 1051 Success: True

{'reading_1': 10.3841571, 'reading_2': 13.30724032, 'reading_3': 21.77096367, 'reading_4': 21.110705600000003, 'reading_5': 20.98199781, 'reading_6': 12.00949679}
======SUCCESS====== 2022-01-04 03:20:28.400029 ===Row: 1052 Success: True

{'reading_1': 10.74583461, 'reading_2': 13.28654255, 'reading_3': 2

2022-01-04 03:22:49,047   wiotp.sdk.device.client.DeviceClient  INFO    Disconnected from the IBM Watson IoT Platform
2022-01-04 03:22:49,051   wiotp.sdk.device.client.DeviceClient  INFO    Closed connection to the IBM Watson IoT Platform


In [8]:
deviceCli.disconnect()

2021-12-07 10:05:20,282   wiotp.sdk.device.client.DeviceClient  INFO    Disconnected from the IBM Watson IoT Platform
2021-12-07 10:05:20,285   wiotp.sdk.device.client.DeviceClient  INFO    Closed connection to the IBM Watson IoT Platform
2021-12-07 10:05:22,037   wiotp.sdk.device.client.DeviceClient  INFO    Connected successfully: d:YOUR_ORGID:YOUR_Device:YOUR_Device001
